app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -64,27 +77,80 @@ internal class DefaultMetricsStorage(
             Event.GrowthData.FirstUriLoadForDay -> {
                 settings.uriLoadGrowthLastSent = System.currentTimeMillis()
             }
+            Event.GrowthData.FirstWeekSeriesActivity -> {
+                settings.firstWeekSeriesGrowthSent = true
+            }
         }
     }
 
+    private fun updateDaysOfUse() {
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData
+        val currentDate = Calendar.getInstance()
+        val asString = dateFormatter.format(currentDate.time)
+        if (currentDate.timeInMillis.withinFirstWeek() && daysOfUse.none { it == asString }) {
+            settings.firstWeekDaysOfUseGrowthData = daysOfUse.plus(asString)
+        }
+    }
+
+    private fun shouldTrackFirstWeekActivity(): Boolean = Result.runCatching {
+        if (!System.currentTimeMillis().withinFirstWeek() || settings.firstWeekSeriesGrowthSent) {
+            return false
+        }
+
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData.map {
+            dateFormatter.parse(it)
+        }.sorted()
+
+        // This loop will check whether the existing list of days of use, combined with the
+        // current date, contains any periods of 3 days of use in a row. It's mostly this ugly
+        // because of the inherent mutability of the Calendar type, but hopefully we can rip
+        // this code out soon™ anyway
+        for (idx in daysOfUse.indices) {
+            if (idx + 1 > daysOfUse.lastIndex || idx + 2 > daysOfUse.lastIndex) {
+                continue
+            }
+            val day1 = daysOfUse[idx]!!.time.toCalendar()
+            val day2 = daysOfUse[idx + 1]!!.time.toCalendar()
+            val day3 = daysOfUse[idx + 2]!!.time.toCalendar()
+            var areSequential = true
+            day1.add(Calendar.DAY_OF_MONTH, 1)
+            areSequential = areSequential && day1 == day2
+            day1.add(Calendar.DAY_OF_MONTH, 1)
+            areSequential = areSequential && day1 == day3
+            if (areSequential) {
+                return true
+            }
+        }
+        return false
+    }.getOrDefault(false)
+
     private fun Long.hasBeenMoreThanDaySince(): Boolean =
         System.currentTimeMillis() - this > dayMillis
 
+    private fun Long.toCalendar(): Calendar = Calendar.getInstance().also {
+        it.timeInMillis = this
+    }
+
+    @Suppress("MagicNumber")
+    private fun Long.withinFirstWeek() = this < getInstalledTime() + dayMillis * 8
+
     companion object {
         private const val dayMillis: Long = 1000 * 60 * 60 * 24
         private const val windowStartMillis: Long = dayMillis * 2
         private const val windowEndMillis: Long = dayMillis * 28
 
         fun shouldSendGenerally(context: Context): Boolean {
```

Comment:
not necessary for the PR, but could this function later get some documentation? Its purpose/result seems ambiguous.